django/db/models/fields/__init__.py

```diff
@@ -812,7 +812,7 @@ def get_choices(self, include_blank=True, blank_choice=BLANK_CHOICE_DASH, limit_
         Return choices with a default blank choices included, for use
         as <select> choices for this field.
         """
-        if self.choices:
+        if self.choices is not None:
```

Comment:
Any required changes in this file should have corresponding test changes. For example, I reverted this change and didn't see any test failures.

Reply:
Not seeing failures makes sense. The change here that matters the most is the change on line 745. The other changes I made mostly out of "correctness".

I guess it begs the question of what should the behavior of field should be if choices is Falsy. The ticket just cares about `get_FOO_display`. 
On the other hand I think it makes sense that if I pass an empty tuple as `choices`, then `get_choices` returns me my empty tuple. The other methods will probably follow as well.

I'll work on adding more tests, and we can see if the API has any semantic change.